lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -164,6 +175,40 @@ def add_descriptor_name(xml, options)
         xml.hps :TxnDescriptor, options[:descriptor_name] if options[:descriptor_name]
       end
 
+      def add_three_d_secure(xml, card_or_token, options)
+        if card_or_token.is_a?(NetworkTokenizationCreditCard)
+          build_three_d_secure(xml, {
+            source: card_or_token.source,
+            cavv: card_or_token.payment_cryptogram,
+            eci: card_or_token.eci,
+            xid: card_or_token.transaction_id,
+          })
+        elsif options[:three_d_secure]
```

Comment:
Should we test the `:three_d_secure` options hash path as well? I didn't see any new or existing tests exercising this logic.

Reply:
Good catch. That'd be a good thing to add. @slogsdon Thoughts?

Reply:
Will do.

Reply:
Done as of 76859d924f9d05cea7aa0115bc219c2a8a0ddacc.

Reply:
Looks great! Thank you @slogsdon

@bayprogrammer Are we good to go on this?